providers/provider_data.go

```diff
@@ -21,3 +25,17 @@ type ProviderData struct {
 
 // Data returns the ProviderData
 func (p *ProviderData) Data() *ProviderData { return p }
+
+func (p *ProviderData) ClientSecret() (ClientSecret string, err error) {
```

Comment:
I wonder if we should call this method `GetClientSecret` and then we can keep the field as `ClientSecret`? WDYT? Any thoughts @syscll @steakunderscore ?

I'm not saying this is a must, just an idea, not sure about having a method and a field so closely named. It's not immediately obvious when you see a field with an underscore at the end of a name that it's part of the field name, if that makes sense?

Reply:
The idea was to reread the file every time we need its content. Because we can't say when exactly the key gets rotated. So it felt like the most straightforward way to go about it - expose the method, discourage the use of the field.

And yeah. It also shows that I've been writing too much Python in the last 5 years 😄
So I'm open to suggestions about the best way to implement it. I understand that my approach here may not be idiomatic for Go.

Reply:
Happy with he behaviour as it is now for what it's worth, just the naming in a little hesitant on

> And yeah. It also shows that I've been writing too much Python in the last 5 years smile
So I'm open to suggestions about the best way to implement it. I understand that my approach here may not be idiomatic for Go.


Haha, I knew I recognised that idiom, just wasn't sure from which language 😅